extensions/amp-live-list/0.1/poller.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {exponentialBackoff} from '../../../src/exponential-backoff';
+
+
+/**
+ * Polling state of a Poller instance.
+ * @enum {string}
+ */
+export const PollerState = {
+  /**
+   * The poller is running at a the set interval.
+   */
+  RUNNING: 'running',
+
+  /**
+   * The poller is running but is doing exponential backoff (using
+   * the original interval value as base) every tick.
+   */
+  BACKEDOFF: 'backedoff',
```

Comment:
Not sure how useful this state is. From a point of view of the poller itself: it can be either running or not running, right? And it can be either in normal mode or recovery mode. But it depends on how you intend the PollerState enum to be used - if it's mostly for internal/testing purposes, it's certainly fine.